### PR TITLE
docs: adopt diataxis for Rockcraft documentation

### PR DESCRIPTION
**Pull request by cjdcordeiro**
This PR addresses the previously planned introduction of the [Diataxis framework](https://diataxis.fr/) into the Rockcraft documentation.

Here are the proposed changes:
 - created a new landing page for the docs
 - added an [How-to page](https://diataxis.fr/how-to-guides/)
 - added an [Explanation page](https://diataxis.fr/explanation/)
 - added a new tutorial
 - minor changes to the existing content in "Reference" and "Tutorials"

**note:** this PR might not contain all the information we could have in the docs at the moment, but said information is more than welcome in future PRs

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Comment by cjdcordeiro** [comment]
ping @valentincanonical

File: docs/explanation.rst
Comment (tigarmo) [inline docs/explanation.rst:30]:
```suggestion
Now, this is where things get interesting. To answer this question, we first need to look at the current state of the art with respect to the existing container image offerings
```

File: docs/explanation.rst
Comment (tigarmo) [inline docs/explanation.rst:45]:
```suggestion
Sure, consumers could venture to fix these vulnerabilities themselves, but not only would this increase the cost and scattering of images, but it wouldn't be 
```

I think we switch the verb and subject order when using 'not only'

Comment (tigarmo) [inline docs/explanation.rst:45]: I think I know what you mean with "scattering", but it seems unclear. Is there a more specific synonym?

File: docs/explanation.rst
Comment (tigarmo) [inline docs/explanation.rst:85]: Maybe we should link to the Snapcraft docs here? They are the most comprehensive ones I think. Something like https://snapcraft.io/docs/snapcraft-overview

File: docs/howto.rst
Comment (tigarmo) [inline docs/howto.rst:69]: There are four (I think) spaces of indentation here, as opposed to the two spaces in the previous command samples. Does that make a difference in the output? 
(yes I agree it's a weird thing to notice)

File: docs/howto.rst
Comment (tigarmo) [inline docs/howto.rst:81]: Should we mention here that currently the only available channel is edge, because Rockcraft is still in development/early stages?

File: docs/index.rst
Comment (tigarmo) [inline docs/index.rst:17]:
```suggestion
**Rockcraft is a tool to create ROCKs** - a new generation of secure, stable and OCI-compliant container images, powered by Ubuntu.
```

**Review by tigarmo — CHANGES_REQUESTED** [review]
Nice! Did the content come from somewhere? Or did you write it all from scratch?
many thanks

File: docs/explanation.rst
Comment (cjdcordeiro) [reply docs/explanation.rst]: proliferation is a better word :) changing

File: docs/howto.rst
Comment (cjdcordeiro) [reply docs/howto.rst:69]: :p fixed and added `sphinx-lint` to the CI

File: docs/howto.rst
Comment (cjdcordeiro) [reply docs/howto.rst:81]: It's mentioned in index.rst that it is in heavy development. I think that's enough. If people follow this how-to they should realize that only edge is available

**Comment by cjdcordeiro** [comment]
> Nice! Did the content come from somewhere? Or did you write it all from scratch? many thanks

thanks ;)

It was a mix. The definitions and concepts were inspired in some of our existing content and presentations. DM me if you'd like access to those

**Review by tigarmo — APPROVED** [review]

**Comment by tigarmo** [comment]
Weird spread failure, I'll re-run it
image

File: docs/explanation.rst
Comment (sergiusens) [inline docs/explanation.rst:50]:
```suggestion
container images vs some of the ones maintained by Canonical, we can verify that the latter have no high/critical vulnerabilities in them!
```

File: docs/howto.rst
Comment (sergiusens) [inline docs/howto.rst:65]: I am not sure why we need multipass

File: docs/tutorials.rst
Comment (sergiusens) [inline docs/tutorials.rst:77]: TODO: we need to humanize this list (we have a method for this)

File: docs/tutorials.rst
Comment (sergiusens) [inline docs/tutorials.rst:112]: docker has parent, skopeo doesn't

File: docs/howto.rst
Comment (cjdcordeiro) [reply docs/howto.rst:65]: isn't it a provider? https://github.com/canonical/rockcraft/blob/main/rockcraft/providers/_multipass.py

(even if one can't really select it atm)

File: docs/tutorials.rst
Comment (cjdcordeiro) [reply docs/tutorials.rst:112]: what do you mean by "parent"?

**Review by sergiusens — APPROVED** [review]
this is a good improvement to what is there, so I am good with it, we can iterate from these solid grounds

File: docs/tutorials.rst
Comment (cjdcordeiro) [reply docs/tutorials.rst]: (I think you meant parenthesis :p if so, that's fixed now)

**Comment by cjdcordeiro** [comment]
thanks. Please wait for @valentincanonical to :+1: before merging.

**Comment by valentincanonical** [comment]
it reads well! thanks @cjdcordeiro for getting this rolling. 👍

File: docs/tutorials.rst
Comment (cjdcordeiro) [reply docs/tutorials.rst:77]: I assume you'll want to create a separate issue for this. resolving for now, to unblock merge status

File: docs/howto.rst
Comment (cjdcordeiro) [reply docs/howto.rst:65]: @sergiusens let me know if it should be removed from the docs

**Comment by sergiusens** [comment]
Let's remove it for now

**Comment by cjdcordeiro** [comment]
> Let's remove it for now

what are you referring to?

**Comment by sergiusens** [comment]
@cjdcordeiro I was referring to the multipass comments, I was hoping each email had an ID to refer back to (it may have and I ruined it by removing the original message :smile: )

**Comment by cjdcordeiro** [comment]
ah :p np, 1 sec

**Comment by sergiusens** [comment]
I would add multipass back in once we explain how to use it

**Comment by cjdcordeiro** [comment]
fair. done

**Pull request merged**
